### ISSUE TITLE
Make runnable Injected be overridable with 'with' block

### ISSUE THREAD

**Issue by proboscis** [proboscis/pinjected#17]

**Comment by proboscis** [comment]
This is a proposal to specify 'overrides' for injected variables defined in a module.

```python
# test_module.py
with providers(x=1):
    y = injected('x')
    with providers(x=2, z=3):
        alpha = injected('x') + injected('z')
```

```bash
% pinjected run test_module.y # -> '1'
% pinjected run test_module.alpha # -> '5'
```

**Comment by proboscis** [comment]
Choices:
```python
with override(providers()):
```
or 
```python
with providers():
```

**Comment by proboscis** [comment]
I prefer `with providers()`, but this will involve modification to Design class, affecting the semantics of the Design.

**Comment by proboscis** [comment]
Implemented with #20 , closing.
